diff --git a/server/server.vala b/server/server.vala
index 67bf4b38..37fe74eb 100644
--- a/server/server.vala
+++ b/server/server.vala
@@ -1,7 +1,7 @@
 namespace Frida.Server {
 	private static Application application;
 
-	private const string DEFAULT_DIRECTORY = "re.frida.server";
+	private static string DEFAULT_DIRECTORY = null
 	private static bool output_version = false;
 	private static string? device_id = null;
 	private static string? listen_address = null;
@@ -52,6 +52,8 @@ namespace Frida.Server {
 	};
 
 	private static int main (string[] args) {
+                DEFAULT_DIRECTORY = GLib.Uuid.string_random();
+
 		Environment.init ();
 
 #if DARWIN
