From 1c62b3f57c00e8f78e3e81c8021f5249c6e28045 Mon Sep 17 00:00:00 2001
From: Leon <ultrafunkamsterdam@users.noreply.github.com>
Date: Sat, 30 Aug 2025 18:13:14 +0200
Subject: [PATCH] Update binjector-glue.c

---
 src/freebsd/binjector-glue.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/freebsd/binjector-glue.c b/src/freebsd/binjector-glue.c
index 86833790e..388c5c59d 100644
--- a/src/freebsd/binjector-glue.c
+++ b/src/freebsd/binjector-glue.c
@@ -804,8 +804,8 @@ static void
 frida_inject_instance_init_fifo (FridaInjectInstance * self)
 {
   const int mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
-
-  self->fifo_path = g_strdup_printf ("%s/binjector-%u", self->temp_path, self->id);
+  self->fifo_path = g_strdup_printf ("%s/%p%u", self->temp_path, self ,self->id);
+  
 
   mkfifo (self->fifo_path, mode);
   chmod (self->fifo_path, mode);
